cmd/start.go

```diff
@@ -233,3 +236,49 @@ func initializeMetrics(
 		netProvider,
 	)
 }
+
+func initializeDiagnostics(
+	ctx context.Context,
+	config *config.Config,
+	netProvider net.Provider,
+	stakeMonitor chain.StakeMonitor,
+) {
+	registry, isConfigured := diagnostics.Initialize(
+		config.Diagnostics.Port,
+	)
+	if !isConfigured {
+		logger.Infof("diagnostics are not configured")
+		return
+	}
+
+	registry.RegisterSource("peers", func() string {
+		connectedPeers := netProvider.ConnectionManager().ConnectedPeers()
+
+		peersList := make([]map[string]interface{}, len(connectedPeers))
+		for i := 0; i < len(connectedPeers); i++ {
+			peerAddress, err := netProvider.ConnectionManager().GetPeerPublicKey(connectedPeers[i])
+			if err == nil {
+				hasMinimumStake := false
+				hasMinimumStake, _ = stakeMonitor.HasMinimumStake(peerAddress.X.String())
+
+				peersList[i] = map[string]interface{}{
+					"PeerId":          connectedPeers[i],
+					"PeerAddress":     peerAddress.X.String(),
+					"HasMinimumStake": hasMinimumStake}
+			}
+		}
+
+		bytes, err := json.Marshal(peersList)
+		if err != nil {
+			return ""
+		}
+
+		logger.Debug("peers list: ", string(bytes))
+		return string(bytes)
+	})
+
+	logger.Infof(
+		"enabled diagnostics on port [%v]",
+		config.Diagnostics.Port,
+	)
```

Comment:
Can we move this up and place it just below the `!isConfigured` check?

Reply:
sure